### PR TITLE
Add static ip to kic/docker on linux

### PR DESCRIPTION
**Pull request by josedonizetti**
fixes https://github.com/kubernetes/minikube/issues/7756

### CODE REVIEW

File: pkg/drivers/kic/oci/oci.go [hunk 1]
```diff
@@ -142,6 +142,13 @@ func CreateContainerNode(p CreateParams) error {
 		"--label", p.NodeLabel,
 	}
 
+	// network
+	if p.OCIBinary == Docker && runtime.GOOS == "linux" { // for now only docker on linux
```
Comment (medyagh) [inline pkg/drivers/kic/oci/oci.go:146]: why only on linux?

**Comment by priyawadhwa** [comment]
Tested locally and this works on linux -- still haven't figured out why it isn't working on my mac.

I ran this script across ~15 iterations and the IP stayed constant on linux:

<details>

```bash
#!/bin/bash


while true; do

minikube stop
minikube start --driver docker
STR=$(minikube ssh sudo systemctl status kubelet)
SUB='node-ip=192.168.39.2'
if [[ "$STR" == *"$SUB"* ]]; then
  echo "IP is what we expect."
else
   echo "Node IP is different"
   exit 1
fi
done
```

</details>

File: pkg/drivers/kic/oci/oci.go (hunk 1 above)
Comment (priyawadhwa) [reply pkg/drivers/kic/oci/oci.go]: I spoke to Jose, and he said:

Docker containers on OSx don’t support ip! It is always localhost
There is why we have the custom tunnel for example

File: test/integration/fn_mount_cmd_test.go [hunk 2]
```diff
@@ -116,7 +116,7 @@ func validateMountCmd(ctx context.Context, t *testing.T, profile string) { // no
 	}
 
 	start := time.Now()
-	if err := retry.Expo(checkMount, time.Millisecond*500, Seconds(15)); err != nil {
+	if err := retry.Expo(checkMount, 5*time.Second, Seconds(15)); err != nil {
```
Comment (medyagh) [inline test/integration/fn_mount_cmd_test.go:119]: I think the 5*Second is the amount of time that waits first time before retrying, again, and 15 is total time. is that what you wanted?

Comment (priyawadhwa) [reply test/integration/fn_mount_cmd_test.go]: ah thanks! accidentally left that in from debugging

**Comment by priyawadhwa** [comment]
@medyagh Linux tests are passing now; the failures on KVM are flakes. i reran the tests and they passed the second time around [logs](https://a646c87040050000000000000000001.proxy.googleprod.com/job/KVM_Linux_integration/11599/consoleFull)

**Pull request merged**